src/haxesharp/collections/Linq.hx

```diff
@@ -38,6 +47,24 @@ class Linq<T>
         return array.indexOf(object) > -1;
     }
 
+    /**
+     *  Returns all the unique elements in the array. Uses indexOf/contains under the hood
+     *  to check for object equality.
+     */
+    public static function distinct<T>(array:Array<T>):Array<T>
+    {
+        var toReturn = new Array<T>();
+        for (item in array)
+        {
+            if (!toReturn.contains(item))

```

Comment:
We might be able to use `Map<K,V>` here to avoid having to call `Array.contains` each iteration of the `item in array` loop, and could instead call `Map.exists` for a lookup that (I think) is better than O(N).  Depends on the implementation of `Map` for the `K` key type, but I think most of them use tree data structures.

The downside with `Map<K,V>` here is we end up having to loop over the `Map.keys()` in the end create the `toReturn` Array.

Reply:
You're right that a map would simplify it. That would increase the space complexity, but that's okay. What I'm more worried about -- as you know -- `Map` doesn't deal with all key types well. While we could use our own `Dictionary`, I think that's overkill.

What I like about this implementation: it's easy to understand. There are Haxe libraries with set support (I couldn't easily find one though) if we decide to go that route.

What do you think?

Reply:
We can always go with this implementation, and if it becomes a CPU bottleneck at any point swap it out for a less understandable but more CPU friendly implementation.

Reply:
So do you want to merge this or do you want me to change it?

Reply:
Merged :+1: